pyrelational/strategies/regression/abstract_regression_strategy.py

```diff
@@ -15,7 +15,7 @@ def __init__(self):
         super(RegressionStrategy, self).__init__()
         self.scoring_fn = NotImplementedError
 
-    def active_learning_step(self, num_annotate: int, data_manager: DataManager, model: ModelManager):
+    def __call__(self, num_annotate: int, data_manager: DataManager, model: ModelManager):
         output = self.train_and_infer(data_manager=data_manager, model=model)
         scores = self.scoring_fn(x=output)
```

Comment:
I think it's safer. Like you could have a flexible scoring_fn that takes different parameters and "x" is not the first one. Not specifying might fail silently, while here it would fail.

Reply:
But in which case it should say in the doc string somewhere that the user defined scoring function must have an input called x? Also for consistency, the x= should be added to the classification_strategy scoring function

Reply:
I am going to hold off on any work here as mypy will require changes and this will all look very different then to satisfy liskov principle for instance... Could you please add this as a comment under the mypy issue though?

Reply:
okay, I'll do that :)